src/main/scala/viper/silver/ast/utility/QuantifiedPermissions.scala

```diff
@@ -98,44 +92,68 @@ object QuantifiedPermissions {
   }
 
   private def quantifiedFields(toVisit: mutable.Queue[Member],
+                               root: Member,
                                collected: mutable.LinkedHashSet[Field],
                                visited: mutable.Set[Member],
                                program: Program): Unit = {
 
     while (toVisit.nonEmpty) {
-      val root = toVisit.dequeue()
+      val currentRoot = toVisit.dequeue()
 
-      root visit {
-        case QuantifiedPermissionAssertion(_, _, acc: FieldAccessPredicate) =>
-          collected += acc.loc.field
-        case Forall(_,triggers,_) => collected ++= triggers flatMap (_.exps) collect {case fa: FieldAccess => fa.field}
+      val relevantNodes: Map[Member, Seq[Node]] = currentRoot match {
+        case m@Method(_, _, _, pres, posts, _) if m != root =>
+          // use only specification of called methods
+          Map(m -> (pres ++ posts))
+        case _ => Map(currentRoot -> Seq(currentRoot))
       }

```

Comment:
isn't the size of the map in `relevantNodes` guaranteed to be 1? If so, we could return a `(Member, Seq[Node])` instead. Additionally, it seems that the key in the map is always equal to `currentRoot`, so we could even return a `Seq[Node]` instead of a tuple